doc/projects.md

```diff
@@ -0,0 +1,142 @@
+- local registry
+- global project
+- local project
+
+The global projects are projects that are automatically provisioned in all
+registries: the global registry hub and all local registries.
+
+The local projects are only provisioned in the configured registries.
+
+Registryman provisions replication rules for global projects using the following
+rule:
+
+** Each repository of a global project is synchronized from the global registry
+hub to the local registries. **
+
+# Realizing Projects
+
+Registryman provides support for multiple registry providers, i.e. multiple
+implementations of the abstract registry resource. Some registry providers (e.g.
+Harbor [2]) implement the concept of projects, some (e.g. ACR [3]) don't.
+
+When the registry provider does not implement the concept of project,
+registryman attempts to emulate the project concept based on the names of the
+repositories.
+
+E.g. in Harbor we have the following projects provisioned:
+
+- `os-images`
+- `databases`
+
+Then we push the `alpine` and `ubuntu` images to the `os-images` project.
+Similarly, we then push the `postgres` and `mongo` images to the `databases`
+project.
+
+In order to do so, we have to tag the images like this:
+
+- `harbor.repo/os-images/alpine`
+- `harbor.repo/os-images/ubuntu`
+- `harbor.repo/databases/postgres`
+- `harbor.repo/databases/mongo`
+
+Sometimes we can't rely on the convenient project feature because the registry
+does not support this concept. In that case we have to push the repositories
+directly to the registry. In that case, we can (and should) came up with a good
+naming strategy for our repositories. 
+For ACR, we could push our 4 images tagged like this:
+
+- `acr.repo/os-images/alpine`
+- `acr.repo/os-images/ubuntu`
+- `acr.repo/databases/postgres`
+- `acr.repo/databases/mongo`
+
+The repository names contain a "namespace" information (i.e.
+`os-images` and `databases`).
+
+This way we can emulate the project concept even if our registry provider does
+not implement it.
+
+Some risks and limitations of the project concept emulation are detailed in the
+next sub-chapters.
+
+## Creation of Projects
+
+It can happen that registryman comes to the conclusion that it shall create a
+new project because the expected state differs from the actual state of a
+registry.
+
+For registry providers with projects that's a meaningful administrative task. 
+
+For registry providers without projects this is impossible to perform. Projects
+are implicitly created when a repository is pushed with the proper tag. For
+example to emulate the creation of the `runtimes` project in ACR, you should
+push a repository called something like `acr.repo/runtimes/jvm`.
+
+## Managing the Members of a Project
+
+With projects we can manage the repository access on project level. We can e.g.
+specify which users have push and which users have only pull capabilities.
+
+Without projects, the repository access can be configured on registry level only.
```

Comment:
I guess that is the case for ACR, but for some registries it may be possible to configure it also on the repository level - e.g. for quay: https://github.com/kubermatic-labs/aquayman/blob/master/config.example.yaml